# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/client/TextureRecorded.cpp
# Commit: 8839a3ba6243
# Full Hash: 8839a3ba62436bd0ea7e5bd07e2b63a8d45b8609
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2023-12-09 09:33:28
# Regressor Bug: 1863914
# File Overlap Count: 1
# Description:
#   Bug 1863914: Use multiple shmem buffers for remote canvas recording. r=aosmond
#   
#   This replaces the use of a single large ring buffer.
#   The buffers are still processed in parallel and are recycled to reduce
#   allocation. Events that do not fit in the default sized buffer have a separate
# ==============================================================================

diff -r f088b7a21045 -r 8839a3ba6243 gfx/layers/client/TextureRecorded.cpp
--- a/gfx/layers/client/TextureRecorded.cpp	Fri Dec 08 15:56:39 2023 +0000
+++ b/gfx/layers/client/TextureRecorded.cpp	Fri Dec 08 16:11:27 2023 +0000
@@ -23,7 +23,7 @@
     already_AddRefed<CanvasChild> aCanvasChild, gfx::IntSize aSize,
     gfx::SurfaceFormat aFormat, TextureType aTextureType)
     : mCanvasChild(aCanvasChild), mSize(aSize), mFormat(aFormat) {
-  mCanvasChild->EnsureRecorder(aTextureType);
+  mCanvasChild->EnsureRecorder(aSize, aFormat, aTextureType);
 }
 
 RecordedTextureData::~RecordedTextureData() {